lib/core/src/display/scene/transform.rs

```diff
@@ -0,0 +1,84 @@
+use nalgebra::{Matrix4, Quaternion, UnitQuaternion, Vector3};
+
+// Note [Quaternion YXZ Euler angles]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// To comply with Threejs impl, we generate a Quaternion applying rotation in
+// the order: pitch -> roll -> yaw, instead of roll -> pitch -> yaw based on
+// https://github.com/mrdoob/three.js/blob/master/src/math/Quaternion.js#L199
+fn from_euler_angles_pry(roll: f32, pitch: f32, yaw: f32) -> UnitQuaternion<f32> {
+    let (s1, c1): (f32, f32) = (roll * 0.5 as f32).sin_cos();
+    let (s2, c2): (f32, f32) = (pitch * 0.5 as f32).sin_cos();
+    let (s3, c3): (f32, f32) = (yaw * 0.5 as f32).sin_cos();
+
+    UnitQuaternion::from_quaternion(Quaternion::new(
+        c1 * c2 * c3 - s1 * s2 * s3,
+        s1 * c2 * c3 + c1 * s2 * s3,
+        c1 * s2 * c3 - s1 * c2 * s3,
+        c1 * c2 * s3 + s1 * s2 * c3,
+    ))
+}
+
+pub struct Transform {
+    pub position:   Vector3<f32>,
+    pub quaternion: UnitQuaternion<f32>,
+    pub scale:      Vector3<f32>,
```

Comment:
`position` and `scale` are the names of physical properties of objects. `quaternion` is a name of a math structure. I may be wrong, but I feel it's inconsistent here.

Reply:
You are right. I renamed `quaternion` to `rotation`. I also renamed `position` to `translation`.